### PR TITLE
Update webui react version and use fluent ui

### PR DESCRIPTION
**Pull request by Lijiaoa**
- update react version
- update fabric ui to fluent ui
- update react-router from v3 to v5
- use `context` to pass value

### CODE REVIEW

File: src/webui/package.json [hunk 1]
```diff
@@ -22,12 +22,16 @@
...
+    "jest-environment-jsdom-fourteen": "0.1.0",
+    "jest-resolve": "24.7.1",
+    "jest-watch-typeahead": "0.3.0",
     "json5": "^2.1.1",
     "mini-css-extract-plugin": "0.5.0",
     "monaco-editor": "^0.19.0",
     "monaco-editor-webpack-plugin": "^1.8.1",
     "node-sass": "^4.13.0",
-    "office-ui-fabric-react": "^7.78.0",
+    "office-ui-fabric-react": "^6.174.0",
```
Comment (ultmaster) [inline src/webui/package.json:34]: Is it still called fabric-react? Or fluent-ui?

File: src/webui/package.json [hunk 2]
```diff
@@ -81,7 +85,7 @@
     "eslint-plugin-react-hooks": "^1.5.0",
     "express": "^4.17.1",
     "npx": "^10.2.0",
-    "typescript": "^3.8.0"
+    "typescript": "3.4.5"
```
Comment (ultmaster) [inline src/webui/package.json:88]: Why is typescript downgraded?

File: src/webui/package.json [hunk 3]
```diff
@@ -63,10 +68,9 @@
     "@types/d3": "^5.7.2",
     "@types/jest": "24.0.12",
     "@types/json5": "^0.0.30",
-    "@types/node": "^10.13.0",
-    "@types/react": "16.8.6",
-    "@types/react-dom": "^16.0.7",
-    "@types/react-router": "^3.0.20",
+    "@types/node": "11.13.8",
```
Comment (ultmaster) [inline src/webui/package.json:71]: Suggest using `^` for versions.

File: src/webui/package.json (hunk 1 above)
Comment (liuzhe-lz) [reply src/webui/package.json:34]: Why it's a downgrade?

**Comment by Lijiaoa** [comment]
@ultmaster @liuzhe-lz 
This PR will change to draft status for continuing to work for fluent ui.

File: src/webui/package.json (hunk 1 above)
Comment (Lijiaoa) [reply src/webui/package.json]: had fixed this issue. now is `"@fluentui/react": "^7.135.0"`

File: src/webui/package.json [hunk 4]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@fluentui/react": "^7.135.0",
```
Comment (ultmaster) [inline src/webui/package.json:6]: Please sort the dependencies alphabetically. (It should be automatic if you had add them with `yarn add`)

Comment (Lijiaoa) [reply src/webui/package.json:6]: `yarn add @fluentui/react` and it is in this line.

Comment (Lijiaoa) [reply src/webui/package.json:6]: @ultmaster by test, @fluentui/react will in "devdependencies"

File: deployment/pypi/install.ps1 [hunk 5]
```diff
@@ -46,7 +46,7 @@ yarn
 yarn build
 Copy-Item config -Destination .\dist\ -Recurse -Force
 cd $CWD\..\..\src\webui
-yarn
+yarn --ignore-engines
```
Comment (liuzhe-lz) [inline deployment/pypi/install.ps1:49]: Why this?

Comment (Lijiaoa) [reply deployment/pypi/install.ps1:49]: to fix error: `error jest-diff@26.4.2：The engine "node" is incompatible with this module. Expected version ">= 10.14.2". Got "10.13.0"`

Comment (liuzhe-lz) [reply deployment/pypi/install.ps1:49]: So why not upgrade node? I'm have been using 10.22.0 in my dev environment and it works fine.

Comment (liuzhe-lz) [reply deployment/pypi/install.ps1:49]: BTW I'm looking forward to bump node to v14 or v12 in nni 2.0.
Node v10 is running out of its life. There's only half a year left.

File: src/webui/package.json (hunk 2 above)
Comment (Lijiaoa) [reply src/webui/package.json]: had revert to ^3.8.0

File: deployment/pypi/install.ps1 (hunk 5 above)
Comment (ultmaster) [reply deployment/pypi/install.ps1:49]: Agreed. Let's bump into at least 10.14.2. 10.22.0 sounds charming to me. 👍

**Review by ultmaster — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**